### PR TITLE
Refactor Enter keypress on Nav Link

### PR DESCRIPTION
**Pull request by jeryj**
## What?

Code quality/cleanup.

## Why?

There are several onClicks within the navigation link block that won't run or are unnecessary.

## How?

Remove/refactor usage.

## Testing Instructions

Should be no change from trunk.

### CODE REVIEW

File: packages/block-library/src/navigation-link/edit.js [hunk 1]
```diff
@@ -220,7 +220,7 @@ export default function NavigationLinkEdit( {
 				hasChildren: !! getBlockCount( clientId ),
 			};
 		},
-		[ clientId ]
+		[ clientId, maxNestingLevel ]
```
Comment (jeryj) [inline packages/block-library/src/navigation-link/edit.js:223]: Linting.

File: packages/block-library/src/navigation-link/edit.js [hunk 2]
```diff
@@ -282,7 +282,7 @@ export default function NavigationLinkEdit( {
 				placeCaretAtHorizontalEdge( ref.current, true );
 			}
 		}
-	}, [ url ] );
+	}, [ url, isLinkOpen, label ] );
```
Comment (jeryj) [inline packages/block-library/src/navigation-link/edit.js:285]: Linting.

File: packages/block-library/src/navigation-link/edit.js [hunk 3]
```diff
@@ -329,10 +329,7 @@ export default function NavigationLinkEdit( {
 	} = getColors( context, ! isTopLevelLink );
 
 	function onKeyDown( event ) {
-		if (
-			isKeyboardEvent.primary( event, 'k' ) ||
-			( ( ! url || isDraft || isInvalid ) && event.keyCode === ENTER )
```
Comment (jeryj) [inline packages/block-library/src/navigation-link/edit.js:334]: This is handled by the `onClick` event that gets added later on. It uses the same checks.

File: packages/block-library/src/navigation-link/edit.js [hunk 4]
```diff
@@ -370,7 +367,9 @@ export default function NavigationLinkEdit( {
 	);
 
 	if ( ! url || isInvalid || isDraft ) {
-		blockProps.onClick = () => setIsLinkOpen( true );
+		blockProps.onClick = () => {
+			setIsLinkOpen( true );
+		};
```
Comment (jeryj) [inline packages/block-library/src/navigation-link/edit.js:376]: onClick handles Enter keypresses, so the onKeydown Enter keypress check is unnecessary.

File: packages/block-library/src/navigation-link/edit.js [hunk 5]
```diff
@@ -518,11 +517,6 @@ export default function NavigationLinkEdit( {
 												'core/image',
 												'core/strikethrough',
 											] }
-											onClick={ () => {
-												if ( ! url ) {
-													setIsLinkOpen( true );
-												}
-											} }
```
Comment (jeryj) [inline packages/block-library/src/navigation-link/edit.js:529]: There is a ! url check on line 478 that makes this section never render if there's no url, so this check will never be valid.

**Review by draganescu — APPROVED** [review]
I am approving this based on all tests passing and the changes making sense on a read.

**Pull request merged**